File: week2/cpp/boxFilter.cpp
/*
Copyright 2017 BIG VISION LLC ALL RIGHTS RESERVED

This code is made available to the students of 
the online course titled "Computer Vision for Faces" 
by Satya Mallick for personal non-commercial use. 

Sharing this code is strictly prohibited without written
permission from Big Vision LLC. 

For licensing and other inquiries, please email 
spmallick@bigvisionllc.com 
*/

#include <opencv2/core/core.hpp>
#include <opencv2/highgui/highgui.hpp>
#include <opencv2/imgproc/imgproc.hpp>
#include <iostream>
#include <cmath>

using namespace cv;
using namespace std;

int main(int argc, char ** argv)
{
  
  string filename = "../data/images/gaussian-noise.png";
  if (argc == 2)
  {   
    filename = argv[1];
  }
    
  Mat image = imread(filename);
  Mat dst1,dst2;

  // Box filter- kernel size 3
  blur( image, dst1, Size( 3, 3 ), Point(-1,-1) );

  //Box filter kernel size 7
  blur(image,dst2,Size(7,7),Point(-1,-1));

  //Display images
  Mat combined;
  cv::hconcat(image, dst1, combined);
  cv::hconcat(combined, dst2, combined);
  namedWindow("Original Image   --   Box Filter Result",CV_WINDOW_AUTOSIZE);
  imshow("Original Image   --   Box Filter Result",combined);  

  imwrite("results/boxBlur3.jpg",dst1);
  imwrite("results/boxBlur7.jpg",dst2);
  waitKey(0);
  destroyAllWindows();
}

File: week2/python/boxFilter.py
# Copyright 2017 BIG VISION LLC ALL RIGHTS RESERVED
# 
# This code is made available to the students of 
# the online course titled "Computer Vision for Faces" 
# by Satya Mallick for personal non-commercial use. 
#
# Sharing this code is strictly prohibited without written
# permission from Big Vision LLC. 
#
# For licensing and other inquiries, please email 
# spmallick@bigvisionllc.com 
# 

import cv2,argparse
import numpy as np

ap = argparse.ArgumentParser()
ap.add_argument("-f", "--filename")
args = vars(ap.parse_args())

filename = "../data/images/gaussian-noise.png"
if args['filename']:
  filename = args['filename']

img = cv2.imread(filename)

# Apply box filter - kernel size 3
dst1=cv2.blur(img,(3,3),(-1,-1))

# Apply box filter - kernel size 7
dst2=cv2.blur(img,(7,7),(-1,-1))

lineType=4
# Scale Factor
fontScale=1

# Display images
combined = np.hstack([img,dst1,dst2])
cv2.namedWindow("Original Image   --   Box Filter Result", cv2.WINDOW_AUTOSIZE)
cv2.imshow("Original Image   --   Box Filter Result",combined)
cv2.waitKey(0)
cv2.destroyAllWindows()
cv2.imwrite("results/boxBlur3.jpg",dst1)
cv2.imwrite("results/boxBlur7.jpg",dst2)
